README.md

```diff
@@ -1,63 +1,166 @@
 # npl-integrations
 
-## Pre-requisites
-Make sure your `hosts` file includes keycloak at 127.0.0.1:
+The npl-integrations repo contains a sample project that demonstrates how to integrate with [Noumena's Engine](https://documentation.noumenadigital.com/engine/applications/engine/) using
+different programming languages and frameworks.
+It includes a Python listener service, a Typescript React web app, and a Python Streamlit UI.
+
+The underlying NPL code is an extension of the IOU, which can be found
+in the [npl-starter](https://github.com/NoumenaDigital/npl-starter) repository.
+
+## Building and running with Docker locally
+
+The project can be built and run locally using Docker Compose.
+To run the Noumena Engine, the corresponding Docker image is required, which is available under license. 
+Please contact Noumena Digital for more information at [info@noumenadigital.com](mailto:info@noumenadigital.com)
+
+The npl-integrations project can be locally built and all containers run locally using Docker Compose.
+
+### Pre-requisites
+
+#### OS X and linux systems
+Make sure the `/etc/hosts` file includes the line `127.0.0.1 keycloak`
 
 ![img.png](docs/img.png)
 
-## Building and running with docker locally
-1. Run `mvn clean install` to build and generate NPL-api and clients.
-2. Run `docker compose up --build -d` to ensure python and frontend are also build, then create and start containers.
+If the line is not present, add it by running `echo "127.0.0.1 keycloak" | sudo tee -a /etc/hosts` in the terminal.
+
+#### Windows systems
+
+Make sure the `c:\Windows\System32\Drivers\etc\hosts` file includes the line `127.0.0.1 keycloak` and add it if missing.
+
+### Build & run
+
+To build and run the project, follow the steps below:
+
+1. Run `mvn clean install` to build and generate NPL-API and clients.
+2. Run `docker compose up --build -d` to ensure the Python listener service, the Python Streamlit UI and the Typescript React frontend are built in addition to the Engine and its dependencies, then create and start the containers.
+
+### Service endpoints
+
+Once the project is running, services can be accessed with the following URLs:
+
+| Service                   | URL                                |
+|---------------------------|------------------------------------|
+| Engine APIs               | http://localhost:12000/            |
+| Swagger UI of Engine APIs | http://localhost:12000/swagger-ui/ |
+| Keycloak admin console    | http://localhost:11000             |
```

Comment:
I believe using http://keycloak:11000 instead of http://localhost:11000 removes redirect errors when you visit that page and klick to access the actual admin console.